internal/tikv/redis_helper/lock.go

```diff
@@ -31,10 +40,15 @@ func (p *RedisPreempt) init() {
 	_, _ = rand.Read(byt)
 	p.password = base64.StdEncoding.EncodeToString(byt)
 	p.lockScript = redis.NewScript(`
-		if redis.call('get',KEYS[1]) == ARGV[1] then
-			return redis.call('expire', KEYS[1], ARGV[2])
-		else
-			return redis.call('set', KEYS[1], ARGV[1], 'ex', ARGV[2], 'nx')
+		local val = redis.call('get', KEYS[1])
+		if (val==nil or (type(val) == "boolean" and not val)) then
+			redis.call('set', KEYS[1], ARGV[1], 'ex', ARGV[2])
+			return 'LockResultSuccess'
```

Comment:
why return text and later return 0 here?  can you return the same kind of value for consistency?

Reply:
This is a two-part script. There are 3 states returned when lock, and only 2 states are returned by unlock, indicating success or failure.